### ISSUE TITLE
Assayapps inline file link creation is limited

### ISSUE THREAD

**Issue by sellth** [bihealth/sodar-server#1625]
My custom assay plugin creates file links for selected text columns, based on code of the meta_ms assay plugin. Due to https://github.com/bihealth/sodar-server/commit/d82da0da78343eec378b2d76ef66d02173132f39 and specifically https://github.com/bihealth/sodar-server/blob/f3e37287b058c4fd6c9db50c46de43de35e3b2d0/samplesheets/rendering.py#L81-L82 this approach only works for columns named "Report file".

This special case treatment is not mentioned in the documentation, not expected, and fails silently. 

### How to Reproduce
Use code like this in assay plugins to create inline file links:

```py
if (
    header['obj_cls'] == 'Process'
    and (
        header['value'].lower() == 'antibody panel'
        or header['value'].lower() == 'barcode key'
        or header['value'].lower() == 'report file'
        )
    and row[i]['value']
):
    row[i]['link'] = (
        base_url + '/' + MISC_FILES_COLL + '/' + row[i]['value']
    )
```
Links are only created in Report file columns. 

### Expected Behavior

Links are created in all matched text columns.

**Comment by mikkonie** [comment]
This is not a bug and doesn't quite work as you say.

The `SPECIAL_FILE_LINK_HEADERS` list contains special cases (well, currently one special case) which are not counted as data file headers, as they are not specified as such in AltamISA / ISA model. We are not causing anything to "only" work with a certain column header, we are extending the functionality to an additional type via an ugly hack. IIRC it was done urgently for some demo.

As long as the column type is in [DATA_FILE_HEADERS](https://github.com/bihealth/altamisa/blob/master/altamisa/constants/table_headers.py#L82), no hack is needed.

Also see:

```
elif (
    name.lower() == 'name' and header['item_type'] == 'DATA'
) or name.lower() in SPECIAL_FILE_LINK_HEADERS:  # HACK for issue #817
    header['col_type'] = 'LINK_FILE'
```

If you wish to add more non-standard hacks like this, we can discuss it. Of course, it would be better to use the build in header names whenever possible.

**Comment by sellth** [comment]
Ok, I had a hard time understanding why my draft plugin did not work as expected and this is what I understood. Thanks for the clarification.

Let's definitely discuss this, as the files I want to link are technically not data files, but part of the metadata.

**Comment by sellth** [comment]
Resolved in d1afebfb738544edc0a908fbd36c70506df8252f.
